src/components/report-viewer-tab.js

```diff
@@ -13,30 +13,83 @@ import { ReportViewer } from '@gridsuite/commons-ui';
+import Switch from '@mui/material/Switch';
+import { useIntl } from 'react-intl';
+import makeStyles from '@mui/styles/makeStyles';
+
+const useStyles = makeStyles((theme) => ({
+    reportOnlyNode: {
+        margin: 5,
+    },
+}));
+
+const NETWORK_MODIFICATION = 'NetworkModification';
 
 /**
  * control the ReportViewer (fetch and waiting)
- * @param reportId : string  uuid of report
+ * @param studyId : string study id
  * @param visible : boolean window visible
+ * @param workingNode : object visualized node
  * @returns {*} node
  * @constructor
  */
-export const ReportViewerTab = ({ reportId, visible }) => {
+export const ReportViewerTab = ({ studyId, visible, workingNode }) => {
+    const intl = useIntl();
+    const classes = useStyles();
+
     const [report, setReport] = useState(null);
     const [waitingLoadReport, setWaitingLoadReport] = useState(false);
     const { enqueueSnackbar } = useSnackbar();
+    const [nodeOnlyReport, setNodeOnlyReport] = useState(true);
+
+    const handleChangeValue = useCallback((event) => {
+        setNodeOnlyReport(event.target.checked);
+    }, []);
 
-    const reportIdRef = useRef();
+    function condenseReport(report) {
+        let newReport = {
+            taskKey: report.taskKey,
+            defaultName: report.defaultName,
+            taskValues: report.taskValues,
+            reports: report.reports,
+            subReporters: [],
+        };
+        report.subReporters.forEach((subReport1) => {
+            if (subReport1.taskKey === NETWORK_MODIFICATION) {
+                // we group all network modifications together
+                subReport1.subReporters.forEach((subReport2) =>
+                    newReport.subReporters.push(subReport2)
+                );
+            } else {
+                newReport.subReporters.push(subReport1);
+            }
+        });
+        return newReport;
+    }
 
     useEffect(() => {
-        if (visible && reportIdRef.current !== reportId) {
+        if (visible) {
             setWaitingLoadReport(true);
-            reportIdRef.current = reportId;
-            fetchReport(reportId)
+            fetchReport(studyId, workingNode.id, nodeOnlyReport)
                 .then((fetchedReport) => {
-                    if (reportIdRef.current === reportId)
-                        // set the report only if it's the last expected/fetched report
-                        setReport(fetchedReport);
+                    if (fetchedReport.length === 1) {
+                        setReport(condenseReport(fetchedReport[0]));
+                    } else {
+                        let globalReport = {
+                            taskKey: 'root',
+                            defaultName: 'Logs',
+                            taskValues: {},
+                            reports: [],
+                            subReporters: [],
+                        };
+                        fetchedReport.forEach((report) => {
```

Comment:
May be `subReporters : fetchedReport.map(condenseReport)` ?
But, may be with horizontal room so expensive, we may gain no clarity. Don't know.

Reply:
Done